packages/base/cards-grid.gts

```diff
@@ -473,7 +483,18 @@ class Isolated extends Component<typeof CardsGrid> {
         },
       });
     });
+
+    this.activeFilter =
+      this.filters.find(
+        (filter) => filter.displayName === this.activeFilter.displayName,
+      ) ?? this.filters[0];
   });
+
+  private refreshFilterList = (ev: MessageEvent) => {
+    if (ev.type === 'index') {
+      this.loadFilterList.perform();
+    }
```

Comment:
you should be more precise with this event, there are 3 variations of this event:
https://github.com/cardstack/boxel/blob/main/packages/runtime-common/realm.ts#L200-L210
- `incremental`
- `full`
- `incremental-index-initiation`

just saying `ev.type ===' 'index'` means that it will always trigger twice for each incremental index, once for `incremental-index-initiation` and once for incremental`. instead you should specifically test for `incremental` so that it will onlky refresh the filter list _after_ the incremental index has been performed.

Reply:
Thanks for the helpful information! I wasn't aware of that.